src/server/editorServices.ts

```diff
@@ -1351,6 +1356,55 @@ namespace ts.server {
             }
         }
 
+        /*@internal*/
+        private updateSharedExtendedConfigFileMap(project: ConfiguredProject) {
+            const extendedSourceFiles = project.getCompilerOptions().configFile?.extendedSourceFiles || emptyArray;
+            extendedSourceFiles.forEach((extendedSourceFile: string) => {
+                const extendedConfigPath = this.toPath(extendedSourceFile);
+                if (!this.sharedExtendedConfigFileMap.has(extendedConfigPath)) {
+                    const watcher = this.watchFactory.watchFile(
+                        extendedConfigPath,
+                        () => this.onSharedExtendedConfigChanged(extendedConfigPath),
+                        PollingInterval.High,
+                        this.hostConfiguration.watchOptions,
+                        WatchType.ExtendedConfigFile
+                    );
+                    this.sharedExtendedConfigFileWatchers.set(extendedConfigPath, watcher);
+                }
+                const otherProjects = this.sharedExtendedConfigFileMap.get(extendedConfigPath);
+                if (!otherProjects || !otherProjects.includes(project)) {
+                    this.sharedExtendedConfigFileMap.add(extendedConfigPath, project);
+                }
+            });

```

Comment:
This also needs to remove older extendedSourceFiles which is not done as part of this.